src/Lykke.Service.PayInternal.Services/BtcTransferRequestService.cs

```diff
@@ -19,18 +21,21 @@ public class BtcTransferRequestService : ITransferRequestService
         private const string BitcoinAssert = "BTC";
         private readonly ITransferRepository _transferRepository;
         private readonly IWalletRepository _walletRepository;
-        private readonly IBitcoinApi _bitcointApiClient;
-        private readonly WalletClient _walletClient;
+        private readonly IBitcoinApiClient _bitcointApiClient;
         private readonly ILog _log;
 
-        #region Ctr
-        public BtcTransferRequestService(ITransferRepository transferRepository, IWalletRepository walletRepository, IBitcoinApi bitcointApiClient, WalletClient walletClient, ILog log)
+        #region .ctors
+        public BtcTransferRequestService(
+            ITransferRepository transferRepository,
+            IWalletRepository walletRepository,
+            IBitcoinApiClient bitcointApiClient,
+            QBitNinjaClient ninjaClient,
+            ILog log)
         {
-            _transferRepository = transferRepository;
-            _walletRepository = walletRepository;
-            _bitcointApiClient = bitcointApiClient;
-            _walletClient = walletClient;
-            _log = log;
+            _transferRepository = transferRepository ?? throw new ArgumentNullException(nameof(transferRepository));
+            _walletRepository = walletRepository ?? throw new ArgumentNullException(nameof(walletRepository));
+            _bitcointApiClient = bitcointApiClient ?? throw new ArgumentNullException(nameof(bitcointApiClient));
+            _log = log ?? throw new ArgumentNullException(nameof(log));
         }
         #endregion
         #region public
```

Comment:
Do we really need initialize SourceAmount array here if you already passing wallets inside CalculateSources?